### PR TITLE
Updated softmax

### PR DESCRIPTION
**Pull request by shivsi**
I have updated to include the softmax implementation. Do suggest edits!

### CODE REVIEW

File: dataset.py [hunk 1]
```diff
@@ -100,31 +100,6 @@ def get_batch(self):
         return balanced_batch_images, balanced_batch_texts
 
 
-def hierarchical_dataset(root, opt, select_data='/'):
-    """ select_data='/' contains all sub-directory of root directory """
-    dataset_list = []
-    dataset_log = f'dataset_root:    {root}\t dataset: {select_data[0]}'
-    print(dataset_log)
```
Comment (zweedao) [inline dataset.py:107]: Remember to remove print statement in production code

File: dataset.py [hunk 2]
```diff
@@ -100,31 +100,6 @@ def get_batch(self):
...
-            select_flag = False
-            for selected_d in select_data:
-                if selected_d in dirpath:
-                    select_flag = True
-                    break
-
-            if select_flag:
-                dataset = LmdbDataset(dirpath, opt)
-                sub_dataset_log = f'sub-directory:\t/{os.path.relpath(dirpath, root)}\t num samples: {len(dataset)}'
-                print(sub_dataset_log)
```
Comment (zweedao) [inline dataset.py:120]: Remember to remove print statement in production code

File: dataset.py [hunk 3]
```diff
@@ -333,3 +308,48 @@ def tensor2im(image_tensor, imtype=np.uint8):
...
     image_pil.save(image_path)
+
+if __name__ == '__main__':
+    """
+    Main method
+    """
+    trainX, trainY = readData(TRAIN_FILENAME) # Training data
+    testX, testY = readData(TEST_FILENAME) # Testing data
+
+    parser = argparse.ArgumentParser()
```
Comment (zweedao) [inline dataset.py:319]: Remember to remove arguments before deploying to production

File: dataset.py [hunk 4]
```diff
@@ -333,3 +308,48 @@ def tensor2im(image_tensor, imtype=np.uint8):
...
+    parser.add_argument("-bs", "--batchSize", dest="batchSize", default=10,
+                        type=int, help="Number of sample in mini-batches")
+    parser.add_argument("-r", "--regStrength", dest="regStrength", default=0.001,
+                        type=float, help="L2 weight decay regularization lambda value")
+    parser.add_argument("-m", "--momentum", dest="momentum", default=0.05,
+                        type=float, help="A momentum value")
+
+    args = parser.parse_args()
+
+    print(
```
Comment (zweedao) [inline dataset.py:333]: remove Print statement before deploying model to production

**Review by zweedao — CHANGES_REQUESTED** [review]

File: dataset.py (hunk 1 above)
Comment (shivsi) [reply dataset.py:107]: Removed print statements!

File: dataset.py (hunk 2 above)
Comment (shivsi) [reply dataset.py:120]: Removed print statements :)

File: dataset.py (hunk 3 above)
Comment (shivsi) [reply dataset.py:319]: We may still need arguments, in order to test on smaller batch files in production. Will be necessary for testing!

File: dataset.py (hunk 4 above)
Comment (shivsi) [reply dataset.py:333]: removed!

File: dataset.py (hunk 3 above)
Comment (zweedao) [reply dataset.py:319]: ok that make sense

**Pull request merged**